src/main/java/com/algorand/algosdk/v2/client/indexer/SearchForApplicationBoxes.java

```diff
@@ -0,0 +1,83 @@
+package com.algorand.algosdk.v2.client.indexer;
+
+import com.algorand.algosdk.v2.client.common.Client;
+import com.algorand.algosdk.v2.client.common.HttpMethod;
+import com.algorand.algosdk.v2.client.common.Query;
+import com.algorand.algosdk.v2.client.common.QueryData;
+import com.algorand.algosdk.v2.client.common.Response;
+import com.algorand.algosdk.v2.client.model.BoxesResponse;
+
+
+/**
+ * Given an application ID, returns the box names of that application sorted
+ * lexicographically.
+ * /v2/applications/{application-id}/boxes
+ */
+public class SearchForApplicationBoxes extends Query {
+
+    private Long applicationId;
+
+    /**
+     * @param applicationId
+     */
+    public SearchForApplicationBoxes(Client client, Long applicationId) {
+        super(client, new HttpMethod("get"));
+        this.applicationId = applicationId;
+    }
+
+    /**
+     * Maximum number of results to return. There could be additional pages even if the
+     * limit is not reached.
+     */
+    public SearchForApplicationBoxes limit(Long limit) {
+        addQuery("limit", String.valueOf(limit));
+        return this;
+    }
+
+    /**
+     * The next page of results. Use the next token provided by the previous results.
+     */
+    public SearchForApplicationBoxes next(String next) {
+        addQuery("next", String.valueOf(next));
```

Comment:
Probably, you'll need to prepend `next` with `b64:`. See my comment in the [python SDK](https://github.com/algorand/py-algorand-sdk/pull/366/files#r948447135).

Reply:
yes, that is a fair point, let's discuss over that thread, and I am leaning towards to add prepend `b64:` to `next`.

Reply:
From the decision we made in https://github.com/algorand/algorand-sdk-testing/pull/219, we keep the `nextToken` method unchanged.